pootle/apps/pootle_app/forms.py

```diff
@@ -80,14 +80,20 @@ def clean_filetypes(self):
                           % filetype))
         return value
 
+    def clean_fullname(self):
+        return self.cleaned_data['fullname'].strip()
+
     def clean_treestyle(self):
         value = self.cleaned_data.get('treestyle', None)
         if not value:
             value = self.instance.treestyle
         return value
 
     def clean_code(self):
-        value = self.cleaned_data['code']
+        value = self.cleaned_data['code'].strip()
+        if not value:
+            raise forms.ValidationError(_("Blank strings cannot be used as a "
```

Comment:
Note that after cleaning if `value` evaluates to `False`, Django will already provide an error message (provided the model has `blank=False`, the default). Given the low chances for this happening, personally I wouldn't add an extra string for localizers as we are already covered.